src/um/iptypes.rs

```diff
@@ -0,0 +1,379 @@
+STRUCT!{struct IP_ADAPTER_GATEWAY_ADDRESS_LH_u_s {
+    Length: ULONG,
+    Reserved: DWORD,
+}}
+UNION!{union IP_ADAPTER_GATEWAY_ADDRESS_LH_u {
+    [u64; 1],
+    Alignment Alignment_mut: ULONGLONG,
+    s s_mut: IP_ADAPTER_GATEWAY_ADDRESS_LH_u_s,
+}}
+STRUCT!{struct IP_ADAPTER_GATEWAY_ADDRESS_LH {
+    u: IP_ADAPTER_GATEWAY_ADDRESS_LH_u,
+    Next: *mut IP_ADAPTER_GATEWAY_ADDRESS_LH,
+    Address: SOCKET_ADDRESS,
+}}
+pub type PIP_ADAPTER_GATEWAY_ADDRESS_LH = *mut IP_ADAPTER_GATEWAY_ADDRESS_LH;
+pub type IP_ADAPTER_GATEWAY_ADDRESS = IP_ADAPTER_GATEWAY_ADDRESS_LH;
+pub type PIP_ADAPTER_GATEWAY_ADDRESS = *mut IP_ADAPTER_GATEWAY_ADDRESS_LH;
+STRUCT!{struct IP_ADAPTER_PREFIX_XP_u_s {
+    Length: ULONG,
+    Flags: DWORD,
+}}
+UNION!{union IP_ADAPTER_PREFIX_XP_u {
+    [u64; 1],
+    Alignment Alignment_mut: ULONGLONG,
+    s s_mut: IP_ADAPTER_PREFIX_XP_u_s,
+}}
+STRUCT!{struct IP_ADAPTER_PREFIX_XP {
+    u: IP_ADAPTER_PREFIX_XP_u,
+    Next: *mut IP_ADAPTER_PREFIX_XP,
+    Address: SOCKET_ADDRESS,
+    PrefixLength: ULONG,
+}}
+pub type PIP_ADAPTER_PREFIX_XP = *mut IP_ADAPTER_PREFIX_XP;
+pub type IP_ADAPTER_PREFIX = IP_ADAPTER_PREFIX_XP;
+pub type PIP_ADAPTER_PREFIX = *mut IP_ADAPTER_PREFIX_XP;
+STRUCT!{struct IP_ADAPTER_DNS_SUFFIX {
+    Next: *mut IP_ADAPTER_DNS_SUFFIX,
+    String: [WCHAR; MAX_DNS_SUFFIX_STRING_LENGTH],
+}}
+pub type PIP_ADAPTER_DNS_SUFFIX = *mut IP_ADAPTER_DNS_SUFFIX;
+pub const IP_ADAPTER_DDNS_ENABLED: DWORD = 0x00000001;
+pub const IP_ADAPTER_REGISTER_ADAPTER_SUFFIX: DWORD = 0x00000002;
+pub const IP_ADAPTER_DHCP_ENABLED: DWORD = 0x00000004;
+pub const IP_ADAPTER_RECEIVE_ONLY: DWORD = 0x00000008;
+pub const IP_ADAPTER_NO_MULTICAST: DWORD = 0x00000010;
+pub const IP_ADAPTER_IPV6_OTHER_STATEFUL_CONFIG: DWORD = 0x00000020;
+pub const IP_ADAPTER_NETBIOS_OVER_TCPIP_ENABLED: DWORD = 0x00000040;
+pub const IP_ADAPTER_IPV4_ENABLED: DWORD = 0x00000080;
+pub const IP_ADAPTER_IPV6_ENABLED: DWORD = 0x00000100;
+pub const IP_ADAPTER_IPV6_MANAGE_ADDRESS_CONFIG: DWORD = 0x00000200;
+STRUCT!{struct IP_ADAPTER_ADDRESSES_LH_u_s {
+    Length: ULONG,
+    IfIndex: IF_INDEX,
+}}
+UNION!{union IP_ADAPTER_ADDRESSES_LH_u {
+    [u64; 1],
+    Alignment Alignment_mut: ULONGLONG,
+    s s_mut: IP_ADAPTER_ADDRESSES_LH_u_s,
+}}
+STRUCT!{struct IP_ADAPTER_ADDRESSES_LH_s {
+    bitfield: ULONG,
+}}
+BITFIELD!{IP_ADAPTER_ADDRESSES_LH_s bitfield: ULONG [
+    DdnsEnabled set_DdnsEnabled[0..1],
+    RegisterAdapterSuffix set_RegisterAdapterSuffix[1..2],
+    Dhcpv4Enabled set_Dhcpv4Enabled[2..3],
+    ReceiveOnly set_ReceiveOnly[3..4],
+    NoMulticast set_NoMulticast[4..5],
+    Ipv6OtherStatefulConfig set_Ipv6OtherStatefulConfig[5..6],
+    NetbiosOverTcpipEnabled set_NetbiosOverTcpipEnabled[6..7],
+    Ipv4Enabled set_Ipv4Enabled[7..8],
+    Ipv6Enabled set_Ipv6Enabled[8..9],
+    Ipv6ManagedAddressConfigurationSupported set_Ipv6ManagedAddressConfigurationSupported[9..10],
+    Reserved set_Reserved[10..32],
+]}
```

Comment:
As this is an *anonymous* union of a single field and a corresponding bitfield, this should follow https://github.com/retep998/winapi-rs/blob/0.3/CONTRIBUTING.md#union-with-a-primitive-field-and-an-anonymous-bitfield-struct-of-the-same-type and have the bitfield applied directly to a `Flags` field in `IP_ADAPTER_ADDRESSES_LH` without the intermediary `IP_ADAPTER_ADDRESSES_LH_s`.